src/App.jsx

```diff
@@ -14,33 +16,103 @@ export const goodsFromServer = [
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button type="button" className="button is-info is-light">
-        Sort alphabetically
-      </button>
+const GoodCard = ({ good }) => <li data-cy="Good">{good}</li>;
 
-      <button type="button" className="button is-success is-light">
-        Sort by length
-      </button>
+const Goodlist = ({ goods }) => (
+  <ul>
+    {goods.map((good, index) => (
+      <GoodCard good={good} />
+    ))}
+  </ul>
+);
 
-      <button type="button" className="button is-warning is-light">
-        Reverse
-      </button>
+export const App = () => {
+  const [selectedColor, setSelectedColor] = useState('');
+  const [reversed, setReversed] = useState(false);
+  const [sortField, setSortField] = useState('');
+  const [showReset, setShowReset] = useState(false);
 
-      <button type="button" className="button is-danger is-light">
-        Reset
-      </button>
-    </div>
+  let visibleGoods = goodsFromServer;
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+  if (selectedColor) {
+    visibleGoods = visibleGoods.filter(good => good.color === selectedColor);
+  }
+
+  if (sortField === 'alphabetically') {
+    visibleGoods = [...visibleGoods].sort((a, b) => a.localeCompare(b));
+  } else if (sortField === 'byLength') {
+    visibleGoods = [...visibleGoods].sort((a, b) => a.length - b.length);
+  }
+
+  if (reversed) {
+    visibleGoods = visibleGoods.reverse();
+  }
+
+  const handleReverseClick = () => {
+    setReversed(!reversed);
+    setShowReset(true);
+  };
+
+  const handleResetClick = () => {
+    setSelectedColor('');
+    setReversed(false);
+    setSortField('');
+    setShowReset(false);
+  };
+
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          type="button"
+          className={classNames('button', 'is-info', {
+            'is-light': sortField !== 'alphabetically',
+          })}
+          onClick={() => {
+            setSortField('alphabetically');
+            setShowReset(true);
+          }}
+        >
+          Sort alphabetically
+        </button>
+
+        <button
+          type="button"
+          className={classNames('button', 'is-success', {
+            'is-light': sortField !== 'byLength',
+          })}
+          onClick={() => {
+            setSortField('byLength');
+            setShowReset(true);
+          }}
+        >
+          Sort by length
+        </button>
+
+        <button
+          type="button"
+          className={classNames('button', 'is-warning', {
+            'is-light': !reversed,
+          })}
+          onClick={handleReverseClick}
+        >
+          Reverse
+        </button>
+
+        {showReset && (
```

Comment:
```suggestion
        {(sortField || reversed) && (
```
you don't need showReset state